tests/integration/test_authd/test_authd_key_request_func.py

```diff
@@ -0,0 +1,154 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check if the 'wazuh-authd' daemon correctly handles the key requests
+       from agents with pre-existing IP addresses or IDs.
+
+tier: 0
+
+modules:
+    - authd
+
+components:
+    - manager
+
+daemons:
+    - wazuh-authd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html
+    - https://documentation.wazuh.com/current/user-manual/registering/key-request.html
+
+tags:
+    - key_request
+'''
+import os
+import shutil
+
+import pytest
+from wazuh_testing.tools import LOG_FILE_PATH, WAZUH_PATH
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.file import read_yaml
+from wazuh_testing.authd import validate_authd_logs
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+
+# Configurations
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+fetch_keys_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'files')
+message_tests = read_yaml(os.path.join(test_data_path, 'test_key_request_messages.yaml'))
+configurations_path = os.path.join(test_data_path, 'wazuh_authd_configuration.yaml')
+configurations = load_wazuh_configurations(configurations_path, __name__, params=None, metadata=None)
+filename = "fetch_keys.py"
+
+shutil.copy(os.path.join(fetch_keys_path, filename), os.path.join("/tmp", filename))
```

Comment:
If this is used in multiple tests, it should be defined as a module scope fixture in test_authd conftest.